File: gluecodium/src/test/resources/smoke/defaults/output/android/com/example/smoke/StructWithEnums.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class StructWithEnums {
    public static final SomethingEnum FIRST_CONSTANT = SomethingEnum.REALLY_FIRST;
    @NonNull
    public SomethingEnum firstField;
    @NonNull
    public SomethingEnum explicitField;
    @NonNull
    public SomethingEnum lastField;
    public StructWithEnums() {
        this.firstField = SomethingEnum.REALLY_FIRST;
        this.explicitField = SomethingEnum.EXPLICIT;
        this.lastField = SomethingEnum.LAST;
    }
    public StructWithEnums(@NonNull final SomethingEnum firstField, @NonNull final SomethingEnum explicitField, @NonNull final SomethingEnum lastField) {
        this.firstField = firstField;
        this.explicitField = explicitField;
        this.lastField = lastField;
    }
}

File: gluecodium/src/test/resources/smoke/defaults/output/cpp/src/smoke/StructWithEnums.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/StructWithEnums.h"
#include <utility>
namespace smoke {
const ::smoke::SomethingEnum StructWithEnums::FIRST_CONSTANT = ::smoke::SomethingEnum::REALLY_FIRST;
StructWithEnums::StructWithEnums( )
{
}
StructWithEnums::StructWithEnums( ::smoke::SomethingEnum first_field, ::smoke::SomethingEnum explicit_field, ::smoke::SomethingEnum last_field )
    : first_field( std::move( first_field ) ), explicit_field( std::move( explicit_field ) ), last_field( std::move( last_field ) )
{
}
}
